cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/system/CDAPLogAppender.java

```diff
@@ -236,7 +236,8 @@ LogPathIdentifier getLoggingPath(Map<String, String> propertyMap) throws Illegal
 
     String namespaceId = propertyMap.get(Constants.Logging.TAG_NAMESPACE_ID);
 
-    if (NamespaceId.SYSTEM.getNamespace().equals(namespaceId)) {
+    if (NamespaceId.SYSTEM.getNamespace().equals(namespaceId)
+      && !propertyMap.containsKey(Constants.Logging.TAG_APPLICATION_ID)) {
```

Comment:
What is this for?

Reply:
log collection, if the logs are from system namespace, there is preconditions check to make sure it has "service-id" whereas if these logs are from application in system namespace, they wouldn't have that.

Reply:
Would be good to put a comment there